testing/test_metal_context.mm

```diff
@@ -71,36 +77,35 @@
 
 TestMetalContext::TextureInfo TestMetalContext::CreateMetalTexture(const SkISize& size) {
   std::scoped_lock lock(textures_mutex_);
-  auto texture_descriptor = fml::scoped_nsobject{
-      [[MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm
-                                                          width:size.width()
-                                                         height:size.height()
-                                                      mipmapped:NO] retain]};
+  MTLTextureDescriptor* texture_descriptor =
+      [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm
+                                                         width:size.width()
+                                                        height:size.height()
+                                                     mipmapped:NO];
 
   // The most pessimistic option and disables all optimizations but allows tests
   // the most flexible access to the surface. They may read and write to the
   // surface from shaders or use as a pixel view.
-  texture_descriptor.get().usage = MTLTextureUsageUnknown;
+  texture_descriptor.usage = MTLTextureUsageUnknown;
 
   if (!texture_descriptor) {
     FML_CHECK(false) << "Invalid texture descriptor.";
     return {.texture_id = -1, .texture = nullptr};
   }
 
   id<MTLDevice> device = (__bridge id<MTLDevice>)GetMetalDevice();
-  sk_cfp<void*> texture = sk_cfp<void*>{[device newTextureWithDescriptor:texture_descriptor.get()]};
-
+  id<MTLTexture> texture = [device newTextureWithDescriptor:texture_descriptor];
   if (!texture) {
     FML_CHECK(false) << "Could not create texture from texture descriptor.";
     return {.texture_id = -1, .texture = nullptr};
   }
 
   const int64_t texture_id = texture_id_ctr_++;
-  textures_[texture_id] = texture;
+  textures_[texture_id] = sk_cfp<void*>{(__bridge void*)texture};
```

Comment:
What is keeping `texture` alive if this is a raw pointer?

Reply:
We wrap the pointer in a sk_cfp<void*> smart pointer and hold onto that via the `textures_` map. sk_cfp does a CFRetain in its ctor.

Reply:
See comment above on why what I wrote is correct, but why we need to do a __bridged_retain.